test/LaunchDarkly.EventSource.Tests/EventSourceTests.cs

```diff
@@ -1,628 +0,0 @@
-        }
-
-        [Fact]
-        public async Task When_reconnecting_the_outgoing_request_overrides_Last_Event_Id_from_configuration()
-        {
-            var lastEventId = "10";
-            var firstResponse = $"id:{lastEventId}\nevent: put\ndata: this is a test message\n\n";
-            var secondResponse = $"id:20\nevent: put\ndata: this is a test message\n\n";
-
-            var handler = new StubMessageHandler();
-            handler.QueueResponse(StubResponse.StartStream(StreamAction.Write(firstResponse), StreamAction.CloseStream()));
-            handler.QueueResponse(StubResponse.StartStream(StreamAction.Write(secondResponse), StreamAction.CloseStream()));
-
-            var configuration = Configuration
-                .Builder(_uri)
-                .HttpMessageHandler(handler)
-                .LastEventId("0")
-                .RequestHeader(Constants.LastEventIdHttpHeader, "0")
-                .Build();
-            var evt = new EventSource(configuration);
-            var first = true;
-            handler.RequestReceived += (s, r) =>
-            {
-                if (first)
-                {
-                    first = false;
-                }
-                else
-                {
-                    evt.Close();
-                }
-            };
-
-            await evt.StartAsync();
-
-            var requests = handler.GetRequests().ToList();
-            Assert.True(requests[0].Headers.Contains(Constants.LastEventIdHttpHeader));
-            Assert.True(requests[1].Headers.Contains(Constants.LastEventIdHttpHeader));
-            var lastEventIdHeader = requests[1].Headers.GetValues(Constants.LastEventIdHttpHeader).ToArray();
-            Assert.Equal(1, lastEventIdHeader.Length);
-            Assert.Equal(lastEventId, lastEventIdHeader[0]);
-        }
-
-        [Fact]
-        public async Task When_Configuration_Request_headers_are_set_then_the_outgoing_request_contains_those_same_headers()
-        {
-            var handler = new StubMessageHandler();
-            handler.QueueResponse(StubResponse.StartStream());
-
-            var headers = new Dictionary<string, string> { { "User-Agent", "mozilla" }, { "Authorization", "testing" } };
-
-            var config = new ConfigurationBuilder(_uri).HttpMessageHandler(handler)
-                .RequestHeaders(headers).Build();
-
-            var evt = new EventSource(config);
-            handler.RequestReceived += (s, r) => evt.Close();
-
-            await evt.StartAsync();
-
-            var request = handler.GetRequests().First();
-            Assert.True(headers.All(
-                    item =>
-                        request.Headers.Contains(item.Key) &&
-                        request.Headers.GetValues(item.Key).Contains(item.Value)
-            ));
-        }
-
-        [Fact]
-        public async Task Given_content_type_not_equal_to_eventstream_when_the_http_response_is_received_then_error_event_should_occur()
```

Comment:
We had some serious sentence oriented programming going on in this file.

Reply:
Yeah, I try not to go too far in the opposite direction, it's possible of course to have overly cryptic short test names, but I do feel like (totally aside from the "complete grammatical sentence" vs. "abbreviated phrase" issue) it's overkill to try to fully describe both the test preconditions and the test assertions in the name of the test. The main purposes of the name of the test are that if the test fails, you can get an idea of what kind of functionality may have failed, and you can easily locate the relevant test code. The test code itself usually shows you what you need to know about the details of the preconditions/assertions (like, we set the content type to X, or we detected event X) or if it can't be made self-evident then code comments serve the same purpose. And concise method names (IMO) actually make it a bit easier to see what kind of tests we have, as a larger proportion of the method name is made up of words that uniquely identify that test.

Anyway, the style differences in this repo are just due to the first version of it having been written fairly quickly a few years ago by a contractor, at a time when we didn't have much of a .NET codebase to compare to.